honeycombio/resource_trigger.go

```diff
@@ -45,6 +45,10 @@ func newTrigger() *schema.Resource {
 				Type:     schema.TypeString,
 				Required: true,
 			},
+			"alert_type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
```

Comment:
We should have the default provided here (`on_change`) and switch this to optional otherwise this will end up being a breaking change (this will also match the documentation above :)).

Another nice user-focused improvement would be to add a `ValidateFunc` using ` validation.StringInSlice()` and checking that it is one of the supported values. Take a look at `filter_combination` in the query specification data source as an example: https://github.com/honeycombio/terraform-provider-honeycombio/blob/main/honeycombio/data_source_query_specification.go#L113-L117